### PR TITLE
feat(sidenav): added event brodcasting

### LINKED ISSUE #1374: mdSidenav: broadcast events
Do you intend to implement any event broadcasting on md-sidenav? With current implementation, -i think- the only way to is understand if a sidenav is opened/closed is to watch a function that returns .isOpen() result. And once you have multiple sidenavs with possible locked states, it is getting difficult to obtain communication with other components. 

What would be pros/cons of adding $rootScope.$broadcast that provides sidebar state information whenever an action is taken?

### PR DESCRIPTION
**Pull request by pepperlk**
Resolves  #1374

[Review on Reviewable](https://reviewable.io/reviews/angular/material/1583)

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
This PR is impacted by considered changes to Angular Material regarding event dispatching and component events.

File: src/components/sidenav/demoBasicUsage/script.js [hunk 1]
```diff
@@ -2,6 +2,21 @@
...
 .controller('AppCtrl', function($scope, $timeout, $mdSidenav, $log) {
+    
+    
+$scope.$on('$mdSideNavOpen', function(e, navname){
+    
+    $log.debug('broadcast for open on: ' +navname);
+    
+});
+    
+    $scope.$on('$mdSideNavClose', function(e, navname){
```
Comment (ilanbiala) [inline src/components/sidenav/demoBasicUsage/script.js]: shouldn't be indented.

File: src/components/sidenav/sidenav.js [hunk 2]
```diff
@@ -273,6 +276,8 @@ function SidenavDirective($timeout, $animate, $parse, $mdMedia, $mdConstant, $co
     function close(ev) {
       ev.preventDefault();
       ev.stopPropagation();
+        
+        $rootScope.$broadcast('$mdSideNavClose', sidenavCtrl.$$mdHandle);
```
Comment (ilanbiala) [inline src/components/sidenav/sidenav.js]: incorrect indentation.

File: src/components/sidenav/demoBasicUsage/script.js [hunk 3]
```diff
@@ -2,6 +2,21 @@
 angular.module('sidenavDemo1', ['ngMaterial'])
 
 .controller('AppCtrl', function($scope, $timeout, $mdSidenav, $log) {
+    
+
```
Comment (ilanbiala) [inline src/components/sidenav/demoBasicUsage/script.js]: Too much spacing probably.

**Comment by ThomasBurleson** [comment]
@jelbourn, @robertmesserle, @rschmukler - I worry this feature would set a dangerous precedent. 
While interim element has promise-based APIs for open/close, this seems to open pandora's box for open/close **events** on dialog, toast, bottomsheet, sideNav, tooltip, etc.

**Comment by jelbourn** [comment]
I'd be hesitant to accept this because Angular 2 does not have an analog to `$broadcast` (a downward event), making migration for ng-material users more difficult if they were to start using this.

**Pull request closed without merging**